src/gui/accountsettings.cpp

```diff
@@ -78,26 +80,26 @@ static const char progressBarStyleC[] =
     "background-color: %1; width: 1px;"
     "}";
 
-bool showEnableE2eeWithVirtualFilesWarningDialog()
+void showEnableE2eeWithVirtualFilesWarningDialog(std::function<void(void)> onAccept)
 {
-    QMessageBox e2eeWithVirtualFilesWarningMsgBox;
-    e2eeWithVirtualFilesWarningMsgBox.setText(AccountSettings::tr("End-to-End Encryption with Virtual Files"));
-    e2eeWithVirtualFilesWarningMsgBox.setInformativeText(AccountSettings::tr("You seem to have the Virtual Files feature enabled on this folder. At "
-                                                                             " the moment, it is not possible to implicitly download virtual files that are "
-                                                                             "End-to-End encrypted. To get the best experience with Virtual Files and"
-                                                                             " End-to-End Encryption, make sure the encrypted folder is marked with"
-                                                                             " \"Make always available locally\"."));
-    e2eeWithVirtualFilesWarningMsgBox.setIcon(QMessageBox::Warning);
-    const auto dontEncryptButton = e2eeWithVirtualFilesWarningMsgBox.addButton(QMessageBox::StandardButton::Ok);
+    const auto e2eeWithVirtualFilesWarningMsgBox = new QMessageBox;
```

Comment:
@FlexW Just a suggestion. You may drop that long name and keep it simple. Something like `messageBox`. This is because it is obvious already from the function name, what this message box is for, and, the instance is not used outside this function.